src/restore.ts

```diff
@@ -1,10 +1,3 @@
-import restoreImpl from "./restoreImpl";
-import { StateProvider } from "./stateProvider";
+import { restoreRun } from "./restoreImpl";
 
-async function run(): Promise<void> {
-    await restoreImpl(new StateProvider());
-}
-
-run();
```

Comment:
This caused a lot of problems in tests - the file was exporting _and_ executing the run function, which feels like an unusual pattern. I moved this into `restoreImpl` and just reference it here directly, which required a bunch of non-functional test reactions.